leftsidebar: Add alert words view to the leftsidebar.

Tirthraj-Raval opened the pull request:
This PR adds alert words message views to the left sidebar as mentioned in the issue description. Also this PR adds alert words settings popover to redirect to the alert word settings.

A new file `left_sidebar_alert_words_popover.hbs` is created for displaying the alert words settings popover. 

Now, as alert words is added to left sidebar, so in the test file I have converted the `is:alerted` narrow to `is_common _narrow` endpoint. 

This pull request adds alert words view to the left sidebar but showing count of unread messages with alert words remain. 

Fixes: #28778 

**Screenshots and screen captures:**

https://github.com/user-attachments/assets/59147382-d8da-4f72-a7e7-8e0ca1bdb993




<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

shuklamaneesh23 commented:
@Tirthraj-Raval , please mark the self-review checklist so that this work can be reviewed.
Also, follow the [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-messages) to ensure issue number should be inlcuded in the commit description.

alya commented:
Please update your commit messages to match the [commit style guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages). "add" is not a useful prefix.

As the PR template says, you will also need to include screenshots, not just a video.

Thanks!

Tirthraj-Raval commented:
In my latest commits I have added the commit description following the commit guidelines. I have added screenshots and videos describing the behavior as instructed by you. Should I reword my previous commits as well? 

Please review this PR.

alya commented:
Please clean up your commit history and post again to request a review. See [here](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) again for guidelines.
